W5500.device.nut

```diff
@@ -0,0 +1,2538 @@
+
+        _spi.write(b);
+
+        (_cs) ? _cs.write(1): _spi.chipselect(0);
+    }
+
+
+
+    // PRIVATE FUNCTIONS
+    // ---------------------------------------------
+
+    // ***************************************************************************
+    // _hexToInt
+    // Returns: an integer representation of the string passed in
+    // Parameters:
+    //      hexString - A string representing a hexadecimal number
+    // **************************************************************************
+    function _hexToInt(hexString) {
+        // Does the string start with '0x'? If so, remove it
+        if (hexString.slice(0, 2) == "0x") hexString = hexString.slice(2);
+
+        // Get the integer value of the remaining string
+        local intValue = 0;
+
+        foreach (character in hexString) {
+            local nibble = character - '0';
+            if (nibble > 9) nibble = ((nibble & 0x1F) - 7);
+            intValue = (intValue << 4) + nibble;
+        }
+
+        return intValue;
+    }
+
+    // ***************************************************************************
+    // _addrToIP
+    // Returns: an array of four numbers representing an ip address or false if it is a string for the DNS
+    // Parameters:
+    //      addr - a string or array representing an IP address
+    // **************************************************************************
+    function _addrToIP(addr) {
+
+        if (typeof addr == "string" && addr.len() > 0) {
+
+            try {
+                local parts = split(addr, ".");
+                if (parts.len() == 4) {
+                    foreach (part in parts) {
+                        local ipart = part.tointeger();
+                        if (ipart.tostring() != part || ipart < 0 || ipart >= 256) {
+                            return false;
+                        }
+                    }
+
+                    return [parts[0].tointeger(), parts[1].tointeger(), parts[2].tointeger(), parts[3].tointeger()];
+                } else {
+                    return false;
+                }
+            } catch (e) {
+                return false;
+            }
+
+        } else if (typeof addr == "blob" && addr.len() == 4) {
+            local parts = array();
+            addr.seek(0, 'b');
+            while (!addr.eos()) {
+                parts.push(addr.readn('b'))
+            }
+            return parts;
+        } else if (typeof addr == "array" && addr.len() == 4) {
+            return addr;
+        } else {
+            throw "Bad IP address";
+        }
+
+    }
+
+
+    // ***************************************************************************
+    // _addrToIP
+    // Returns: an array of four numbers representing an ip address or false if it is a string for the DNS
+    // Parameters:
+    //      addr - a string or array representing an IP address
+    // **************************************************************************
+
```

Comment:
We shouldn't be a) polling here and b) do in in a blocking way so that the imp can't process any external events and requests. Just doesn't seem the right thing to do.